scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1106,15 +1113,33 @@ def _update_cluster_name(self, k8s):
         finally:
             self._lock.release()
 
-    def _update_api_server_version(self, k8s):
-        """Update the API server version"""
-        gitver = k8s.get_api_server_version()
+    def _update_api_server_version_if_necessary(self, k8s, current_time=None):
+        """Update the API server version if it has not been successfully update in the last hour.
+        The version number is determined by querying the K8s API server."""
+
+        if current_time is None:
+            current_time = time.time()
+
+        # Check if we have the version set and what time we set it.
         self._lock.acquire()
         try:
-            self._api_server_version = gitver
+            is_version_set = self._api_server_version is not None
+            last_check_time = self._last_api_server_version_update
         finally:
             self._lock.release()
 
+        # We only update if we haven't updated it in the last hour.
+        if is_version_set or current_time - last_check_time > 3600:
```

Comment:
Should it be `if not is_version_set` here?

Reply:
Argh, good catch.  This is why we need unit tests for this code.  The smoke test passes because this just results in us constantly resetting the version which does not impact correctness.